src/App.tsx

```diff
@@ -1,19 +1,176 @@
 import React from 'react';
-import './App.css';
+import classNames from 'classnames';
+import { TodoList } from './components/TodoList/TodoList';
+
+import { User } from './types/User';
+import { TodoPrepared } from './types/TodoPrepared';
+
+import './App.scss';
 
 import users from './api/users';
+import todos from './api/todos';
+
+const preparedTodos = todos.map(todo => {
+  const user = users.find(person => person.id === todo.userId);
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+  return {
+    user: user || null,
+    ...todo,
+  };
+});
+
+type State = {
+  usersCopy: User[],
+  todosCopy: TodoPrepared[],
+  todoTitle: string,
+  selectedUserId: number,
+  selectError: boolean,
+  titleError: boolean,
 };
 
+class App extends React.Component<{}, State> {
+  state: State = {
+    usersCopy: [...users],
+    todosCopy: [...preparedTodos],
+    todoTitle: '',
+    selectedUserId: 0,
+    selectError: false,
+    titleError: false,
+  };
+
+  handleValidation = () => {
+    if (!this.state.todoTitle) {
+      this.setState({
+        titleError: true,
+      });
+    } else if (!this.state.selectedUserId) {
+      this.setState({
+        selectError: true,
+      });
+    }
+
+    if (!this.state.selectedUserId || !this.state.todoTitle) {
+      return false;
+    }
+
+    return true;
+  };
+
+  handleAddTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      todoTitle: event.target.value,
+    });
+  };
+
+  handleSelecte = (event :React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      selectedUserId: Number(event.target.value),
+    });
+  };
+
+  handleSubmit = (event: React.SyntheticEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!this.handleValidation()) {
+      return;
+    }
+
+    this.setState((state) => ({
+      todoTitle: '',
+      selectedUserId: 0,
+      selectError: false,
+      titleError: false,
+      todosCopy: [
+        ...state.todosCopy,
+        {
+          userId: state.selectedUserId,
+          id: state.todosCopy.length + 1,
```

Comment:
id is a maximal id in the array + 1
What if tomorrow will you add the delete method?